### PR TITLE
sqlite: update to 3.38.1.

### PR DESCRIPTION
**Pull request by ologantr**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture (x86_64-glibc)

### CODE REVIEW

File: common/shlibs [hunk 1]
```diff
@@ -352,7 +352,7 @@ libpoppler-cpp.so.0 poppler-cpp-0.18.2_1
 libpoppler-qt5.so.1 poppler-qt5-0.31.0_1
 libtcl8.6.so tcl-8.6.0_1
 libtk8.6.so tk-8.6.1_3
-libsqlite3.so.0 sqlite-3.38.0_1
+libsqlite3.so.0 sqlite-3.38.1_1
```
Comment (paper42) [inline common/shlibs:355]: Why is this necessary?

Comment (ologantr) [reply common/shlibs:355]: I was trying to avoid making it lag behind. 
If it's unnecessary I can revert this bit.

Comment (paper42) [reply common/shlibs:355]: common/shlibs list shlibs and packages in which they are in versions in which they first appeared. `xbps-src` will scan the package for required shlibs, look at this file and create a list of dependencies. Without this change, a new package/an updated one will depend on sqlite>=3.38.0, with this change it will depend on sqlite>=3.38.1.
If there is no good reason to do this, we shouldn't do it. This makes reverts hard and prevents partial updates. It would make sense if sqlite added some new symbols and packages built with this version would crash with the old one. Then we would want new packages (built with new sqlite) to also depend on the new sqlite version, but that's not this case.
